frappe/database/query.py

```diff
@@ -474,25 +537,37 @@ def set_fields(self, fields, **kwargs):
 		if is_str:
 			if fields == "*":
 				return fields
-			if " as " in fields:
-				fields, reference = fields.split(" as ")
-				fields = Field(fields).as_(reference)
+			if "`" in fields:
+				fields = PseudoColumn(fields)
+			if " as " in str(fields):
```

Comment:
a bit confused over here

since if we find backticks in fields then we convert it to pseudocolumn and then we look for ` as ` and then again we check if backticks are present in stringified fields

also the query gets a bit changed (notice the "as" in 2nd one):
<img width="842" alt="Screenshot 2022-09-01 at 1 52 54 AM" src="https://user-images.githubusercontent.com/32034600/187775046-ce1545bc-7c77-4754-9291-362939ea70ec.png">

Reply:
all grave quote fields need to be converted to pseudocolumns and aliasing has a different logic all togather